engine/render/src/render/font_renderer.cpp

```diff
@@ -1038,26 +1038,36 @@ namespace dmRender
         HRenderContext render_context = (HRenderContext)params.m_UserData;
         TextContext& text_context = render_context->m_TextContext;
 
+        // This function is called for both game object text (labels) and gui
+        // Once the sprites are done rendering, it is ok to reuse/reset the state
+        // and the gui can start rendering its text
+        // See also ClearRenderObjects() in render.cpp for clearing the text entries
+        if (text_context.m_Frame != text_context.m_PreviousFrame)
```

Comment:
It's better to let the render engine decide when a new frame occurs.
We already had this concept for the same reason.